docs/agent_api.rst

```diff
@@ -3,4 +3,313 @@
+
+------
+
+.. code-block:: cpp
+
+    bool create_client(uint32_t key, uint8_t session, uint16_t mtu, Middleware::Kind middleware_kind, OpResult& op_result);
+
+Allows to create a `ProxyClient` entity, which can act on behalf of an external client to request the creation/deletion of DDS entities.
+
+Returns `true` if the creation was successful, or `false` otherwise.
+
+:key: The `ProxyClient`'s identifier.
+:session: The session ID to which the created `ProxyClient` is attached to.
+:mtu: The *Maximum Transmission Unit* size.
+:middleware_kind: The middleware used by the `ProxyClient`, to be chosen among the ones presented in the :ref:`middleware_abstraction_layer`.
+:op_result: The result status of this operation.
+
+------
+
+.. code-block:: cpp
+
+    bool delete_client(uint32_t key, OpResult& op_result);
+
+Deletes a certain `ProxyClient` from the client proxy database, given its ID.
+
+Returns `true` if the operation was completed successfully, or `false` otherwise (for example, if the given ID was not registered to any `ProxyClient`).
+
+:key: The identifier of the `ProxyClient` to be removed.
+:op_result: The result status of the operation.
+
+------
+
+.. code-block:: cpp
+
+    bool create_participant_by_xml(uint32_t client_key, uint16_t participant_id, int16_t domain_id, const char* xml, uint8_t flag, OpResult& op_result);
+
+Creates a DDS participant for a certain client, given its self-contained description in a XML file.
+
+The participant will act the entrypoint for the rest of DDS entities to be created.
+
+Returns `true` if the creation was successful, or `false` otherwise.
+
+:client_key: The identifier of the `ProxyClient` to which the resulting participant will be attached to.
+:participant_id: The identifier of the participant to be created.
+:domain_id: The DDS domain ID associated to the participant.
+:xml: The XML describing the participant properties.
+:flag: It determines the creation mode of the new participant (see :ref:`creation_mode_table`).
+:op_result: The result status of this operation.
+
+------
+
+.. code-block:: cpp
+
+    bool create_participant_by_ref(uint32_t client_key, uint16_t participant_id, int16_t domain_id, const char* ref, uint8_t flag, OpResult& op_result);
+
+Creates a DDS participant for a certain client, given a reference to its description hosted in a certain XML descriptor file.
+
+The references file must have been previously loaded by the agent.
+
+The participant will act the entrypoint for the rest of DDS entities to be created.
+
+Returns `true` if the creation was successful, or `false` otherwise.
+
+:client_key: The identifier of the `ProxyClient` to which the resulting participant will be attached to.
+:participant_id: The identifier of the participant to be created.
+:domain_id: The DDS domain ID associated to the participant.
+:ref: The reference tag which will retrieve the participant description from the references file, previously loaded by the agent.
```

Comment:
```suggestion
:ref: The reference tag which will retrieve the participant description from the reference file, previously loaded to the *Agent*.
```

Reply:
It is not a reference file, it is a file containing references. Therefore I think it is clearer to keep "references file" or  "from the file where the references are defined"